fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersUtil.java

```diff
@@ -79,185 +64,95 @@
     private static final String COMMA = ",";
     private static final String EQUALS = "=";
 
-    // Unsupported Operations in FHIR Path
-    public static final String OPERATION_RESOLVE = "resolve()";
-    public static final String OPERATION_CONTAINS = ".contains";
-    private static final List<String> UNSUPPORTED_OPERATIONS = Arrays.asList(OPERATION_CONTAINS);
-
-    private static final String REPLACE_RESOLVE = "(\\.where\\([\\s]{0,}resolve\\(\\)[\\s]{0,}[is]{0,2}[\\s]{0,}[\\w]{0,}\\))";
+    private static final Map<String, ParametersMap> builtInSearchParameters = loadBuiltIns();
 
-    private static final String CONTAINS_EXPR = "\\.contains\\.";
-    private static final String CONTAINS_EXPR_REPLACE = ".`contains`.";
-    private static final String CONTAINS_END_EXPR = ".contains";
-    private static final String CONTAINS_END_EXPR_REPLACE = ".`contains`";
-
-    private static final Map<String, Map<String, SearchParameter>> builtInSearchParameters = loadBuiltIn();
-
-    private static final String INVALID_EXPRESSION = "/NONE/";
+    private static final String MISSING_EXPRESSION = "/NONE/";
 
     private ParametersUtil() {
         // No Operation
     }
 
     /**
-     * Loads the class in the classloader to initialize static members. Call this before using the class in order to
-     * avoid a slight performance hit on first use.
+     * Loads the class in the classloader to initialize static members.
+     * Call this before using the class in order to avoid a slight performance hit on first use.
      */
     public static void init() {
         // No Operation
     }
 
-    /*
-     * wraps the error thrown, catches it, and enables a user to operate.
-     * @return
-     */
-    private static Map<String, Map<String, SearchParameter>> loadBuiltIn() {
-        Map<String, Map<String, SearchParameter>> result = null;
-        try {
-            result = populateSearchParameterMapFromResource(FHIR_DEFAULT_SEARCH_PARAMETERS_FILE);
-        } catch (IOException e) {
-            // This Exception is HIGHLY improbable.
-            log.warning(BUILTIN_ERROR_EXCEPTION);
-        }
-        return result;
-    }
-
     /**
-     * This is a convenience function that simply returns the built-in (spec-defined) SearchParameters as a map keyed by
-     * resource type.
+     * Loads the built-in search parameters and constructs . 
+     * 
+     * @return a map of ParametersMaps, keyed by resourceType
+     * @throws FHIRParserException 
      */
-    public static Map<String, Map<String, SearchParameter>> getBuiltInSearchParameterMap() {
-        return builtInSearchParameters;
-    }
+    private static Map<String, ParametersMap> loadBuiltIns() {
```

Comment:
If we're in a full refactoring mode, I'd suggest extrapolating the relevant signatures to an interface/factory, and the containing class/object into a interface so future changes are opaque to the user.

Reply:
The diff on this one isn't great (maybe due to the large scope of changes).  We already had a private method called loadBuildIn.  I also preserved a method similar to getBuiltInSearchParameterMap but renamed it to buildSearchParametersMapFromBundle.

As you can see, I didn't really attempt to preserve consistency with the past...just went with whatever felt right to me.  I'm happy to try preserving as much of the former API as desired if thats what we want, although the big change here is that the return types will have the new ParametersMap instead of the old `Map<String, SearchParameter>`.

Reply:
No worries.  just a point to consider.  No action required.